backports.cpp

```diff
@@ -62,3 +63,9 @@ LLVMMetadataRef LLVMGoDIBuilderCreateCompileUnit(
     SplitNameLen, Kind, DWOId, SplitDebugInlining, DebugInfoForProfiling);
 #endif
 }
+
+void LLVMGoAddConstantPropagationPass(LLVMPassManagerRef PM) {
+#if LLVM_VERSION_MAJOR < 12
+  LLVMAddConstantPropagationPass(PM);
+#endif
+}
```

Comment:
I think this function should be removed entirely and update TinyGo to not use the constant propagation pass (I believe the SCCP pass is a good replacement).

Reply:
OK, did that here and in `tinygo`.

Reply:
The LLVM12 release notes are (partially) out now, and they say to use `InstSimplify`; should we do that instead?

Reply:
Oh, interesting. Might be worth trying.

However, I've replaced this pass with IPSCCP so this isn't necessary for the switch to LLVM 12.